pymc/model.py

```diff
@@ -1101,6 +1102,36 @@ def add_coords(
         for name, values in coords.items():
             self.add_coord(name, values, length=lengths.get(name, None))
 
+    def set_dim(self, name: str, new_length: int, coord_values: Optional[Sequence] = None):
```

Comment:
```suggestion
    def resize_dim(self, name: str, new_length: int, coord_values: Optional[Sequence] = None):
```

Reply:
I named it `set_dim` to align with the related method `set_data` and because it's not always resizing the dimension, but can also be used to just change coords.

Reply:
Then perhaps we can make the docstrings a bit more in line with it?